### DISCUSSION TITLE
org.signal.libsignal.zkgroup.InvalidInputException: Deserialization failure in zkgroup (AssertionError)

### DISCUSSION THREAD

**Discussion by Rosch** [AsamK/signal-cli#1424] [Q&A] [answered]
Hi..

Just updated from signal-cli 0.11.2 to 0.12.7 on linux. Now I receive an error when using it

```
[www@sproxy ~]$ /usr/local/bin/signal-cli -o json -u MYNUMBER receive
Error loading state file for user MYNUMBER: org.signal.libsignal.zkgroup.InvalidInputException: Deserialization failure in zkgroup (AssertionError)
```
I am running on Oracle Linux 8 which comes with glibc 2.28 so I needed to replace the bundled libsignal with a version compatible to glibc 2.28. I used the one from https://github.com/exquo/signal-libs-build/releases/tag/libsignal_v0.39.3 (by putting it in the libsignal-client jar)

What do I need to do else?

Thanks for your help

Roland

**Comment by AsamK** [comment, marked as answer]
See  #1419

**Reply by Rosch** [reply]
thanks. Now it is working.

So the version of the .so must match the version number of the .jar
